spec/controllers/http_status_agent_spec.rb

```diff
@@ -36,6 +36,9 @@ def f.get url
 
                        def f.set url, response, time = nil
                          sleep(time/1000) if time
+                         def response.to_hash
```

Comment:
@darrencauthon that's fine too, although I might define a method to return the struct instead of repeating `Struct.new(:status, :headers, :to_hash)` a lot.  If you'd like to give this a go, I'll merge @strugee's PR, then you can have at it. No rush!